apps/varnish.go

```diff
@@ -52,3 +53,29 @@ func (r MetricsReceiverVarnish) Pipelines() []otel.Pipeline {
 func init() {
 	confgenerator.MetricsReceiverTypes.RegisterType(func() confgenerator.Component { return &MetricsReceiverVarnish{} })
 }
+
+type LoggingProcessorVarnish struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+}
+
+func (LoggingProcessorVarnish) Type() string {
+	return "varnish"
```

Comment:
What do you think of having this be a receiver instead? And we would have the files receiver and the processor implemented here in the PR. That way the user won't have to configure a files receiver to use this processor.

Reply:
The reason we didn't initially create a varnish receiver is that there is no default log location for varnish. The varnish docs lay out the process for [configuring varnishncsa](https://docs.varnish-software.com/tutorials/enabling-logging-with-varnishncsa/) which is what we're generally following in the integration tests for varnish in order to validate logging.

As there is no default log location, we thought the right path was to provide the processor for parsing, but not pre-define a receiver. If that's not the right path, we're happy to change it.

Reply:
I think for consistency with the other receivers we should go with this being modelled as a receiver. We could default to reading from `/var/log/varnish/varnishncsa.log` and explicitly call that out in our docs. If the user chooses to configure varnish to log to other locations, they can always update the `include_paths` param in the receiver.

There also seems to be some prior art for this. Our agent should document the configuration requirements that it requires from Varnish to work as intended and that should be enough.

Reply:
-1, I don't think we should invent a path unless that's actually something that's commonly used out there.

Do you think `/var/log/varnish/varnishncsa.log` is actually a common path? Do any of the distros ship sample configs to generate it, for instance?

Reply:
The log file mentioned is from their official docs here: https://docs.varnish-software.com/tutorials/enabling-logging-with-varnishncsa/

And there is prior art for this pattern and this log file specifically. See https://github.com/DataDog/integrations-core/tree/master/varnish#log-collection for an example.

Maybe there is more research required to see what the common log file is but this path seems to be the only one I've found being used

Reply:
Synced offline, moving forward with `/var/log/varnish/varnishncsa.log` is okay since there is some convention